cmd/mt-whisper-importer-reader/main.go

```diff
@@ -248,123 +204,118 @@ func shortAggMethodString(aggMethod whisper.AggregationMethod) string {
 	}
 }
 
-func getMetric(w *whisper.Whisper, file string) (*archive.Metric, error) {
+func getMetrics(w *whisper.Whisper, file string) (archive.Metric, error) {
+	var res archive.Metric
 	if len(w.Header.Archives) == 0 {
-		return nil, errors.New(fmt.Sprintf("ERROR: Whisper file contains no archives: %q", file))
+		return res, fmt.Errorf("Whisper file contains no archives: %q", file)
 	}
 
-	archives := make([]archive.Archive, 0, len(w.Header.Archives))
-	var chunkSpan uint32
-	var rowKey string
-	var aggMethodStr string
+	var archives []archive.Archive
 	name := getMetricName(file)
 
-	// md gets generated from the first archive in the whisper file
-	md := getMetricData(name, int(w.Header.Archives[0].SecondsPerPoint))
-
-	for archiveIdx, archiveInfo := range w.Header.Archives {
-		if archiveIdx > 0 {
-
-			// on the first aggregation we determin the aggregation method as string
-			if aggMethodStr == "" {
-				aggMethodStr = shortAggMethodString(w.Header.Metadata.AggregationMethod)
-
-				if aggMethodStr == "" {
-					return nil, errors.New(fmt.Sprintf(
-						"ERROR: Aggregation method in file %s not allowed: %d(%s)\n",
-						file,
-						w.Header.Metadata.AggregationMethod,
-						aggMethodStr,
-					))
-				}
-			}
-
-			rowKey = api.AggMetricKey(
-				md.Id,
-				aggMethodStr,
-				archiveInfo.SecondsPerPoint,
-			)
-		} else {
-			rowKey = md.Id
-		}
-
-		// only read archive if archiveIdx is in readArchives
-		if _, ok := readArchives[archiveIdx]; !ok && len(readArchives) > 0 {
-			continue
-		}
+	if !nameFilter.Match([]byte(name)) {
```

Comment:
why is this filter used so late? seems like we can do the filtering before we even open/read the file, and before we even put the filename on the channel.

Reply:
Changed that in the latest commit

Reply:
why is it still after opening file and after putting it on channel?

Reply:
moved it to before pushing the name into the chan